### PR TITLE
Add Error Prone check: ZoneIdOfZ

### PR DESCRIPTION
**Pull request by ksobolew**
There is a convenient static field which can replace use of a magic String constant of "Z". In fact, at least in the JDK I'm using, these objects are exactly the same:

```
$ jshell
|  Welcome to JShell -- Version 11.0.10
|  For an introduction type: /help intro

jshell> var z = java.time.ZoneId.of("Z");
z ==> Z

jshell> var utc = java.time.ZoneOffset.UTC;
utc ==> Z

jshell> z == utc;
$3 ==> true
```

### CODE REVIEW

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 public class TimestampDecoder
         implements Decoder
 {
-    private static final ZoneId ZULU = ZoneId.of("Z");
+    private static final ZoneId ZULU = ZoneOffset.UTC;
```
Comment (findepi) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java:38]: static import UTC and inline below

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 2]
```diff
@@ -147,6 +147,12 @@
     private final ZoneOffset fixedOffsetEast = ZoneOffset.ofHoursMinutes(2, 17);
     private final ZoneOffset fixedOffsetWest = ZoneOffset.ofHoursMinutes(-7, -31);
 
+    private final ZoneId zoneIdGmt = ZoneId.of("GMT");
+    private final ZoneId zoneIdUtc = ZoneId.of("UTC");
+    @SuppressWarnings("ZoneIdOfZ") // there's an equivalent static field, but we'd like to test it anyway
+    private final ZoneId zoneIdZ = ZoneId.of("Z");
```
Comment (findepi) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:153]: Is it equivalent or "mostly equivalent"?
It surely describes the same zone, but is it indiscernible?
For example GMT, UTC and Z are _not_  completely the same thing.

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 3]
```diff
@@ -1384,10 +1390,10 @@ public void testTimestampWithTimeZone(boolean insertWithTrino)
             tests.addRoundTrip(dataType, afterEpoch.atZone(kathmandu));
             tests.addRoundTrip(dataType, afterEpoch.atZone(fixedOffsetEast));
             tests.addRoundTrip(dataType, afterEpoch.atZone(fixedOffsetWest));
-            tests.addRoundTrip(dataType, afterEpoch.atZone(ZoneId.of("GMT")));
-            tests.addRoundTrip(dataType, afterEpoch.atZone(ZoneId.of("UTC")));
```
Comment (findepi) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1388]: `ZoneId.of("UTC")` occurs a number of times across the code base (and, `ZoneId.of("UTC")` is not the same thing as `ZoneOffset.UTC` and cannot be used interchangeably). Is it also being flagged by ZoneIdOfZ check, or this is an unrelated change?

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java (hunk 2 above)
Comment (ksobolew) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:153]: I wasn't actually sure, perhaps I'm too trusting in what ErrorProne says :) So I ran this:

```
jshell 
|  Welcome to JShell -- Version 11.0.10
|  For an introduction type: /help intro

jshell> var z = java.time.ZoneId.of("Z");
z ==> Z

jshell> var utc = java.time.ZoneOffset.UTC;
utc ==> Z

jshell> z == utc;
$3 ==> true

jshell> 
```

So it looks like they are exactly the same, at least in the JDK version I tried.

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java (hunk 3 above)
Comment (ksobolew) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1388]: This is unrelated, in that I wanted to make it possible to attach the `@SuppressWarnings` annotation on something smaller than the entire method, so I refactored it slightly in spirit of other pieces of this class. Yes, I noticed that some of this is duplicated, but it's in `static` methods, which don't have access to instance fields (all the other fields are also non-`static` so I made these new ones to match for consistency).

Comment (ksobolew) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1388]: I backed out these changes, as I think now there's no point in additionally testing the "Z" zone, as the test is already including `UTC`.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java (hunk 1 above)
Comment (ksobolew) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java]: Sure, done

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java (hunk 3 above)
Comment (findepi) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1388]: if ZoneId.of("Z") is the same as ZoneOffset.UTC, then it's not the same as ZoneId.of("UTC")

i remember some date/time formatting being different for ZoneId.of("UTC") and ZoneOffset.UTC

Comment (ksobolew) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1388]: Yes, and I don't claim that `ZoneId.of("UTC")` should be replaced. Anyway, this change is gone now, so...

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 4]
```diff
@@ -1386,7 +1386,6 @@ public void testTimestampWithTimeZone(boolean insertWithTrino)
             tests.addRoundTrip(dataType, afterEpoch.atZone(fixedOffsetWest));
             tests.addRoundTrip(dataType, afterEpoch.atZone(ZoneId.of("GMT")));
             tests.addRoundTrip(dataType, afterEpoch.atZone(ZoneId.of("UTC")));
-            tests.addRoundTrip(dataType, afterEpoch.atZone(ZoneId.of("Z")));
```
Comment (findepi) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1389]: Use ZoneOffset.UTC instead or removing

**Review by findepi — APPROVED** [review]

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java (hunk 4 above)
Comment (ksobolew) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:1389]: It's already there, at line 1383
